src/discordtipbot/presence_cache.cr

```diff
@@ -0,0 +1,30 @@
+# Stores all online users
+class PresenceCache
+  getter users : Set(UInt64) = Set(UInt64).new
+
+  def delete(user : Discord::PartialUser)
+    users.delete(user.id)
+  end
+
+  def add(user : Discord::PartialUser)
+    users.add(user.id)
+  end
+
+  def online?(user : UInt64)
+    users.includes?(user)
+  end
+
+  def handle_presence(payload : Array(Discord::Presence))
+    payload.each do |x|
+      handle_presence(x)
+    end
+  end
+
+  def handle_presence(presence : Discord::Presence | Discord::Gateway::PresenceUpdatePayload)
+    if presence.status == "online"
```

Comment:
I think this should also include dnd.

Reply:
The idea here really is, to only soak (store) users who are actually online

Reply:
DnD isn't "offline" though; I ride DnD all the time while I'm online, and I know a bunch of other people who do it too. I think it's pretty common.

Reply:
Yeah. I know. IIRC you don't go AFK though if you're in DnD, and it's supposed to only track people who are actively using discord (or a selfbot or sth)

(Using the word `online` in this context might not be that good of an idea, but I actually only want to track people who are marked as `online` by discord